nomos-da/network/subnetworks-assignations/src/lib.rs

```diff
@@ -1,10 +1,25 @@
 pub mod versions;
 
-use std::{collections::HashSet, hash::Hash};
+use std::{
+    collections::{HashMap, HashSet},
+    hash::Hash,
+    sync::Arc,
+};
 
 use libp2p::Multiaddr;
 use libp2p_identity::PeerId;
 
+pub trait MembershipCreator: MembershipHandler {
```

Comment:
Why does the creator also have to be a handler itself? Could it not return a handler as an associated type instead?

Reply:
I think it would be also easier to mock and unit test

Reply:
That's a good suggestion, will update